### ISSUE TITLE
Images in Content Collections are not worked in Astro v5

### ISSUE THREAD

**Issue by psephopaiktes** [withastro/astro#12772]
### Astro Info

```block
Astro                    v5.0.9
Node                     v20.12.0
System                   macOS (arm64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             @astrojs/mdx
                         @astrojs/sitemap
```

### Describe the Bug

From v5, I can no longer access images placed in the same folder as mdx files in Content Collections.

Here is the usage from the official documentation:
- v4: [https://v4.docs.astro.build/en/guides/images/#images-in-content-collections](https://v4.docs.astro.build/en/guides/images/#images-in-content-collections)
- v5: [https://docs.astro.build/en/guides/images/#images-in-content-collections](https://docs.astro.build/en/guides/images/#images-in-content-collections)

I have created a sample here.
https://github.com/psephopaiktes/astro-images-in-content-collections

Most of it remains the same as the [official blog sample](https://github.com/withastro/astro/tree/main/examples/blog) for each version, but I have changed the following files:
- /src/content/config.ts (in v5, it's /src/content.config.ts)
- Directory structure for blog images
- Frontmatter of blog Markdown

In v4, it works fine as per the documentation, but in v5, the images either return a 404 error or the following error occurs:

> A `getStaticPaths()` route pattern was matched, but no matching static path was found for requested path `/blog/first-post/[object%20Object]`

The reason I want this structure is that I want to place the images for each blog post in the same folder as the post `mdx` file.

Bad:
- src/content/blog/first-post.md
- src/public/img/first-post-cover.jpg

Good:
- src/content/blog/first-post/index.md
- src/content/blog/first-post/cover.md

### What's the expected result?

Images placed in the same location as the .md file should be handled as Content Collections without errors.

If there is another solution to place images in the same folder, that would be OK to me.

### Link to Minimal Reproducible Example

https://github.com/psephopaiktes/astro-images-in-content-collections

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by psephopaiktes** [comment]
@ascorbic Thank you for your response!

> the `Image` component from `astro:assets`

I thought it was impossible because it requires dynamically importing images from the Content Collection using this method. However, by referring to the following document, I was able to solve it!

[Dynamically import images | Docs](https://docs.astro.build/en/recipes/dynamically-importing-images/)

For those who encountered the same problem, I will also share the resolved configuration.

[astro-images-in-content-collections/v5-2 at main · psephopaiktes/astro-images-in-content-collections](https://github.com/psephopaiktes/astro-images-in-content-collections/tree/main/v5-2)

---

BTW, even with the code as in the official documentation, same error occurs.

https://docs.astro.build/en/guides/images/#images-in-content-collections

In other words, the content of the official documentation is incorrect, and the relevant section should be removed from the documentation?
